test/com/google/enterprise/adaptor/database/UniqueKeyTest.java

```diff
@@ -248,16 +312,18 @@ public void testProcessingDocId() throws Exception {
 
   @Test
   public void testProcessingDocId_allTypes() throws Exception {
-    executeUpdate("create table data(c1 integer, c2 bigint, c3 varchar(20), "
-        + "c4 date, c5 time, c6 timestamp)");
-    executeUpdate("insert into data(c1, c2, c3, c4, c5, c6) "
-        + "values (123, 4567890, 'foo', "
+    executeUpdate("create table data(c1 integer, c2 bigint, c3 numeric(9,2), "
```

Comment:
Yes. It's just the auto-detect that's broken on H2, because it returns Types.DECIMAL, which we don't want to support (it's inexact). Not sure why they chose that for BigDecimal, maybe just got confused by the names. Numeric types work just fine as long as you declare them as such.